### PR TITLE
Consistent interface for pixel center

### PR DESCRIPTION
**Pull request by varunagrawal**
Defined u0 and v0 in all camera models for consistent interface.

### CODE REVIEW

**Review by dellaert — CHANGES_REQUESTED** [review]
There is already px and py ?? I do not support to duplicate versions of all these things

**Comment by varunagrawal** [comment]
> There is already px and py ?? I do not support to duplicate versions of all these things

`Cal3Bundler` doesn't have `px` or `py` so this causes undefined function errors when working with different camera models (e.g. using `Cal3_S2` in the main code and then loading from a `Cal3Bundler` from a BAL file. I figured having `u0` and `v0` made more sense, though I can make Cal3Bundler have `px` & `py` instead?

**Comment by swarrier246** [comment]
> > There is already px and py ?? I do not support to duplicate versions of all these things
> 
> `Cal3Bundler` doesn't have `px` or `py` so this causes undefined function errors when working with different camera models (e.g. using `Cal3_S2` in the main code and then loading from a `Cal3Bundler` from a BAL file. I figured having `u0` and `v0` made more sense, though I can make Cal3Bundler have `px` & `py` instead?

Wouldn't it be advisable to remove the px and py in the other models and reflect those changes across all places that use it, then? Might cause some avoidable confusion otherwise.

File: gtsam/gtsam.i [hunk 1]
```diff
@@ -597,6 +597,7 @@ class Rot3 {
   Rot3(double R11, double R12, double R13,
       double R21, double R22, double R23,
       double R31, double R32, double R33);
+  Rot3(double w, double x, double y, double z);
```
Comment (swarrier246) [inline gtsam/gtsam.i:600]: Is this related to this PR? (I'm assuming it's a quaternion rot constructor?)

**Comment by varunagrawal** [comment]
> > > There is already px and py ?? I do not support to duplicate versions of all these things
> > 
> > 
> > `Cal3Bundler` doesn't have `px` or `py` so this causes undefined function errors when working with different camera models (e.g. using `Cal3_S2` in the main code and then loading from a `Cal3Bundler` from a BAL file. I figured having `u0` and `v0` made more sense, though I can make Cal3Bundler have `px` & `py` instead?
> 
> Wouldn't it be advisable to remove the px and py in the other models and reflect those changes across all places that use it, then? Might cause some avoidable confusion otherwise.

That would break backwards compatibility. Also a lot of internal code uses the `px` and `py` functions.

File: gtsam/gtsam.i (hunk 1 above)
Comment (varunagrawal) [reply gtsam/gtsam.i:600]: It isn't, but it was minor and I didn't want to create a whole new PR for it.

**Comment by dellaert** [comment]
Please request a re-review after adding px and py to missing Cal. Remove all u0 and v0 that were not there yet. Pls deprecate u0 and v0 if there was any.

File: gtsam/geometry/Cal3Bundler.h [hunk 2]
```diff
@@ -98,11 +98,23 @@ class GTSAM_EXPORT Cal3Bundler {
...
+  inline double px() const {
+    return u0_;
+  }
+
+  /// image center in y
+  inline double py() const {
+    return v0_;
+  }
+
+  /// @deprecated
```
Comment (dellaert) [inline gtsam/geometry/Cal3Bundler.h:111]: No, we have this #ifdef thingy, remember ? :-)

File: gtsam/gtsam.i [hunk 3]
```diff
@@ -2761,7 +2762,16 @@ virtual class EssentialMatrixFactor : gtsam::NoiseModelFactor {
 };
 
 #include <gtsam/slam/dataset.h>
+
 class SfmTrack {
+  SfmTrack();
+
+  double r;
```
Comment (dellaert) [inline gtsam/gtsam.i:2769]: double r,g,b;

File: gtsam/gtsam.i [hunk 4]
```diff
@@ -2761,7 +2762,16 @@ virtual class EssentialMatrixFactor : gtsam::NoiseModelFactor {
...
+  double g;
+  double b;
+  //TODO Need to close wrap#10 to allow this to work.
```
Comment (dellaert) [inline gtsam/gtsam.i:2772]: Please use Google-style linting, which should flag an issue here.

**Review by dellaert — APPROVED** [review]
Thanks, pls address comments though.

**Review by swarrier246 — COMMENTED** [review]
LGTM!

File: gtsam/geometry/Cal3Bundler.h (hunk 2 above)
Comment (varunagrawal) [reply gtsam/geometry/Cal3Bundler.h]: Uhhh I'm stupid

**Comment by varunagrawal** [comment]
Will squash and merge once CI passes.

File: gtsam/gtsam.i (hunk 3 above)
Comment (varunagrawal) [reply gtsam/gtsam.i]: The wrap parser doesn't support this syntax. :(
